help/scan.c

```diff
@@ -53,10 +53,9 @@ static int add_file(const char *fpath, const struct stat *sb, int tflag, struct
   if (tflag == FTW_F)
   {
     char *ext = strrchr(fpath, '.');
-    char *path = mutt_str_strdup(fpath);
     if (ext && !mutt_str_strcmp(ext, ".md"))
     {
-      vector_new_append(&DocPaths, sizeof(char *), path);
+      vector_new_append(&DocPaths, sizeof(char *), mutt_str_strdup(fpath));
```

Comment:
Only copy the path if we're going to use it.
Before, any unused paths were copied and leaked.